File: android/src/main/java/com/facebook/flipper/testing/FlipperResponderMock.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.flipper.testing;

import com.facebook.flipper.core.FlipperArray;
import com.facebook.flipper.core.FlipperObject;
import com.facebook.flipper.core.FlipperResponder;
import java.util.LinkedList;
import java.util.List;

public class FlipperResponderMock implements FlipperResponder {
  public final List<Object> successes = new LinkedList<>();
  public final List<FlipperObject> errors = new LinkedList<>();

  @Override
  public void success(FlipperObject response) {
    successes.add(response);
  }

  @Override
  public void success(FlipperArray response) {
    successes.add(response);
  }

  @Override
  public void success() {
    successes.add(new FlipperObject.Builder().build());
  }

  @Override
  public void error(FlipperObject response) {
    errors.add(response);
  }
}

File: iOS/FlipperKitTestUtils/FlipperResponderMock.m
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "FlipperResponderMock.h"

@implementation FlipperResponderMock

- (instancetype)init {
  if (self = [super init]) {
    _successes = @[];
    _errors = @[];
  }
  return self;
}

- (void)success:(NSDictionary*)response {
  _successes = [_successes arrayByAddingObject:response];
}

- (void)error:(NSDictionary*)response {
  _errors = [_errors arrayByAddingObject:response];
}

@end
